packages/collector/test/tracing/cloud/aws-sdk/v3/kinesis/test_definition.js

```diff
@@ -0,0 +1,239 @@
+/*
+ * (c) Copyright IBM Corp. 2023
+ */
+
+'use strict';
+
+const { v4: uuid } = require('uuid');
+const { cleanup } = require('./util');
+const semver = require('semver');
+const { checkStreamExistence } = require('./util');
+const { expect } = require('chai');
+const path = require('path');
+const { fail } = expect;
+const supportedVersion = require('@instana/core').tracing.supportedVersion;
+const config = require('../../../../../../../core/test/config');
+const { retry, stringifyItems, delay } = require('../../../../../../../core/test/test_util');
+const ProcessControls = require('../../../../../test_util/ProcessControls');
+const globalAgent = require('../../../../../globalAgent');
+const {
+  verifyHttpRootEntry,
+  verifyExitSpan,
+  verifyHttpExit
+} = require('@instana/core/test/test_util/common_verifications');
+const { promisifyNonSequentialCases } = require('../promisify_non_sequential');
+let streamName = process.env.AWS_KINESIS_STREAM_NAME || 'nodejs-team';
+
+if (process.env.AWS_KINESIS_STREAM_NAME) {
+  streamName = `${process.env.AWS_KINESIS_STREAM_NAME}${semver.major(process.versions.node)}-${uuid()}`;
+}
+const withErrorOptions = [true, false];
+
+const operationsInfo = {
+  deleteStream: 'deleteStream',
+  createStream: 'createStream',
+  getRecords: 'getRecords',
+  getShardIterator: 'shardIterator',
+  listStreams: 'listStreams',
+  putRecord: 'putRecord',
+  putRecords: 'putRecords'
+};
+
+const requestMethods = ['async', 'promise', 'cb', 'async-v2', 'promise-v2', 'cb-v2'];
+const availableOperations = [
+  'createStream',
+  'putRecords',
+  'putRecord',
+  'listStreams',
+  'getRecords',
+  'getShardIterator'
+];
+
+const getNextCallMethod = require('@instana/core/test/test_util/circular_list').getCircularList(requestMethods);
+function start(version) {
+  let mochaSuiteFn;
+  if (!supportedVersion(process.versions.node)) {
```

Comment:
Question: I think we need to skip the test on Node v14?
See https://github.com/aws/aws-sdk-js-v3/blob/v3.379.1/clients/client-kinesis/package.json#L74

You skipped it already in the cleanup function. I am wondering why we need to check this in the cleanup function at all?

Reply:
The missing node version check has been added. Concerning the node version check in the cleanup method, all other similar instruments have it, which is why I included it. We can remove it if it is no longer required 🙂

Reply:
Yeah it would be cool if you could double check that, because the Node version check in the cleanup fn is confusing IMO.

Reply:
Removed the check from cleanup function.